acceptancetests/assess_updateseries.py

```diff
@@ -0,0 +1,307 @@
+        raise AssertionError(
+            "Series from {} doesn't match expected: {}".format(
+                unit, series_two))
+
+    try:
+        # response = remote.run(
+        #     'sudo juju-updateseries --to-series {} --from-series {}'.format(
+        #         series_two, series_one))
+        output = client.get_juju_output('ssh', unit,
+                    'sudo juju-updateseries --to-series {} --from-series {}'.format(series_two,
+                                                                                    series_one))
+    except subprocess.CalledProcessError as e:
+        raise JujuAssertionError(
+            "error running juju-updateseries on {}: {}".format(unit, e))
+    # add verify code here
+    log.info(
+        "juju-updateseries ... succeeded: {}".format(output))
+
+    try:
+        # response = remote.run(
+        #     'sudo juju-updateseries --to-series {} --from-series {}' +
+        #     '--start-agents'.format(series_two, series_one))
+        output = client.get_juju_output('ssh', unit,
+                                        'sudo juju-updateseries --to-series {} --from-series {} --start-agents'.format(series_two,
+                                                                                                        series_one))
+    except subprocess.CalledProcessError as e:
+        if reboot in 'after':
+            log.info(
+                'sudo juju-updateseries ... --start-agents: {}'.format(
+                    output))
+            # Where did the output go???
+            # stdout_details = json.loads(output[0]['Stdout'])
+            # if 'systemd is not fully running, please reboot to start agents' \
+            #         not in stdout_details:
+            #     raise JujuAssertionError(
+            #         "error running juju-updateseries on {}: {}".format(
+            #             unit, e))
+        else:
+            raise JujuAssertionError(
+                "error running juju-updateseries on {}: {}".format(unit, e))
+
+    if reboot in 'after':
+        reboot_unit(client, unit, True)
+        # remote.timeout = 420  # 8 minutes
+        # try:
+        #     remote.run('sudo shutdown -r now')
+        # except subprocess.CalledProcessError as e:
+        #     raise AssertionError("shutdown -r now on {}: {}".format(unit, e))
+        # wait for reboot
+        # client.wait_for_started()
+
+    # add verify code here
+    client.wait_for_started()
+    log.info("juju-updateseries .... --start-agents succeeded")
+
+
+def reboot_unit(client, unit, expect_agents):
+    """ Reboot a unit, wait for the agents to restart, or if not agents
+    restarting is not expected, wait for the ssh port to be available.
+    """
+    try:
+        log.info("Restarting unit: {}".format(unit))
+        # client.juju('run', ('--unit', unit, 'sudo shutdown -r now'))
+        output = client.get_juju_output('ssh', unit, 'sudo shutdown -r now && exit', timeout=300)
```

Comment:
Try something like subprocess.run(["ssh", "", "sudo shutdown -r now && exit"]).

If that fails, use subprocess.popen.

you don't need to run with get_juju_output -- we don't care about the output in this case